### ISSUE TITLE
swift-development branch crashes in 32-bit simulator 

### ISSUE THREAD

**Issue by codeflows** [ReactiveCocoa/ReactiveCocoa#1888]
Hi,

I recently updated to Xcode 6.3 (Swift 1.2) and started playing with the latest version of the swift-development branch (built via Carthage). I'm now getting crashes running this piece of code (there's nothing else in my sample project) in the iOS simulator:

```swift
    let producer: SignalProducer<Int, NoError> = SignalProducer { observer, _ in
      sendNext(observer, 1)
      sendCompleted(observer)
    }

    producer |> collect |> start(
      next: { x in println("Next value \(x)") },
      completed: { _ in println("Completed") }
    )
```

The crashes (EXC_BAD_ACCESS, https://gist.github.com/codeflows/2a63b97ef7e360d262d4) don't happen on each run, and not always in the same place. But when they do happen, it's only on the 32-bit simulators (iPhone 4s, 5) but not on the 64-bit ones (iPhone 5s, 6).

Curious about this, I opened the ReactiveCocoa Xcode project and ran the unit tests, and they were also crashing with e.g. iPhone 5:

![screenshot 2015-04-12 04 37 07](https://cloud.githubusercontent.com/assets/29481/7104050/67d514c8-e0ce-11e4-9315-111ec18c5797.png)

Can anybody else reproduce this?

- Xcode 6.3 (6D570)
- Carthage 0.6.4
- Cartfile.resolved: github "ReactiveCocoa/ReactiveCocoa" "6fddd9498b5fd30a3e50c9f7085ffd3d2740dcff"

**Comment by NachoSoto** [comment]
I wonder if this is the same as http://www.openradar.me/20167826

**Comment by codeflows** [comment]
Not sure, but I'm running the tests (and my own code) with compiler optimizations disabled (`-Onone`).

**Comment by jspahrsummers** [comment]
This sounds worthy of a radar. It should be impossible to cause memory management crashes in Swift code, so the fault probably lies with the compiler or the runtime.

Of course, finding a workaround in RAC would be nice, but I would want the underlying issue to be fixed upstream.

**Comment by codeflows** [comment]
I could try narrowing this down and creating a radar later today.

**Comment by codeflows** [comment]
Radar filed: http://www.openradar.me/20523979

**Comment by jspahrsummers** [comment]
@codeflows Thank you! :bow:

**Comment by codeflows** [comment]
Got a pingback from Apple asking if this happens when running on a device as well - tried quickly on my iPhone 5 and got a crash, so I'm following up with them as soon as I have time (next week.)

**Comment by codeflows** [comment]
I sent Apple an updated project that crashes on iPhone 5 (actual device). The nasty thing is that the crashes are not reproducible 100% of the time, there's something random going on.

**Comment by ikesyo** [comment]
It might run correctly now by the fixes in #1905 (fixes for operators underlying the `collect` implementation).

> Curious about this, I opened the ReactiveCocoa Xcode project and ran the unit tests, and they were also crashing with e.g. iPhone 5:

The test is also using `collect` operator.

**Comment by codeflows** [comment]
Yes! Can confirm both my own project and RAC tests seem to be crash-free now. :) :trophy:
